### PR TITLE
Adapt bundle manager to manage dynamic relationships

### PR DESCRIPTION
**Pull request by azdagron**
- Run now reloads configuration from source on a 10 second interval
- A new RefreshBundle call is added (satisfying the interface from #2513) that refreshes the bundle for the given trust domain. It reloads the configuration beforehand.

Fixes: #2514

### CODE REVIEW

**Review by amartinezfayo — COMMENTED** [review]
Looking great.

File: pkg/server/bundle/client/updater.go [hunk 1]
```diff
@@ -70,40 +83,53 @@ func (u *bundleUpdater) UpdateBundle(ctx context.Context) (*bundleutil.Bundle, *
...
-func (u *bundleUpdater) TrustDomainConfig() TrustDomainConfig {
-	return u.c.TrustDomainConfig
+func (u *bundleUpdater) GetTrustDomainConfig() TrustDomainConfig {
+	u.trustDomainConfigMtx.Lock()
+	trustDomainConfig := u.trustDomainConfig
+	u.trustDomainConfigMtx.Unlock()
+	return trustDomainConfig
+}
+
+func (u *bundleUpdater) SetTrustDomainConfig(trustDomainConfig TrustDomainConfig) bool {
```
[the remark this answers is not in the document]
Comment (azdagron) [reply pkg/server/bundle/client/updater.go:101]: Hmm, this should have been covered by the config refresh test, but looks like I missed something. I'll shore this up.

[the remark this answers is not in the document]
Comment (azdagron) [reply pkg/server/bundle/client/updater.go:101]: Oh, yeah. The manager tested only the interface. This function needs a concrete test. I'll add it.

File: pkg/common/telemetry/names.go [hunk 2]
```diff
@@ -436,7 +442,10 @@ const (
 	// Type tags a type
 	Type = "type"
 
-	// TrustDomainID tags some trust domain ID
+	// TrustDomainName tags the name of some trust domain
+	TrustDomainName = "trust_domain_id"
```
[the remark this answers is not in the document]
Comment (azdagron) [reply pkg/common/telemetry/names.go:446]: Good catch :) I renamed it to TrustDomain = "trust_domain"

File: pkg/server/bundle/client/manager.go [hunk 3]
```diff
@@ -39,27 +44,81 @@ type EndpointProfileInfo interface {
 
 type HTTPSWebProfile struct{}
 
+func (p HTTPSWebProfile) Name() string {
+	return "https_web"
```
[the remark this answers is not in the document]
Comment (azdagron) [reply pkg/server/bundle/client/manager.go:48]: Hmm, I don't think so? While I'd like them to remain consistent, one is intended as a switching value in the datastore and the other for a metric label?

[the remark this answers is not in the document]
Comment (marcosy) [reply pkg/server/bundle/client/manager.go:48]: that's a good point! 👍🏻

**Review by amartinezfayo — APPROVED** [review]
LGTM! Thanks @azdagron!

**Pull request merged**